### PR TITLE
Improve detail of exporter addition docs

### PR DESCRIPTION
**Pull request by theotherjimmy**

### CODE REVIEW

**Comment by theotherjimmy** [comment]
@AnotherButler, your assistance in all matters editing is appreciated.

**Comment by theotherjimmy** [comment]
@sg- @toyowata @screamerbg @maclobdell This is the documentation improvement I promised.

**Comment by theotherjimmy** [comment]
@AnotherButler @iriark01 I'm pretty sure that this document is not referenced from anywhere in other part of the contribution guide. Should we add a few more references to it?

File: docs/cont/adding_exporters.md [hunk 1]
```diff
@@ -1,18 +1,26 @@
 # Adding exporters 
 
-This is a guide for adding exporters to the mbed-os tools. It covers the structure of the export subsystem and the individual exporter.
+This is a guide for adding exporters to the mbed-os tools. First, This document describes what an exporter is and what rules it follows. Then, it covers the structure of the export subsystem and the individual exporter. Finally, this document gives some implementation suggestions.
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please lower case "this" in the second sentence.

File: docs/cont/adding_exporters.md [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
 
 <span class="notes">**Note:** All paths are relative to [https://github.com/ARMmbed/mbed-os/](https://github.com/ARMmbed/mbed-os/).</span>
 
+## What is an exporter
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please add a question mark after "exporter" or rewrite as "What an exporter is"

File: docs/cont/adding_exporters.md [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+
+An exporter is a python plugin to the mbed OS tools that convert a project using mbed CLI into one specialized for a particular IDE. For the best user experience, an exporter should:
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please capitalize "Python".

File: docs/cont/adding_exporters.md [hunk 4]
```diff
@@ -230,3 +238,89 @@ $(PROJECT).elf: $(OBJECTS) $(SYS_OBJECTS) $(LINKER_SCRIPT)
...
+### GNU ARM Eclipse
+
+If your IDE uses Eclipse and uses the GNU ARM Eclipse plugin, then you should 
+specialize or alias your exporter with the generic GNU ARM Eclipse.
+
+#### Alias
+
+If you do not need any specialization of the export, then replace your
+exporters class in the `EXPORT_MAP` with the `GNUARMEclipse` class. For example,
+if KDS met all of these requirements, we could:
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please change "we" to "you" for consistent person.

File: docs/cont/adding_exporters.md [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
 
+## What is an exporter
+
+An exporter is a python plugin to the mbed OS tools that convert a project using mbed CLI into one specialized for a particular IDE. For the best user experience, an exporter should:
+ - Take input from the resource scan.
+ - Use the flags in the build profiles.
+ - Have a single template file for each file type they produce. For example, an eclipse CDT project would have one template for `.project` files and one for `.cproject` files.
+ - Not call mbed CLI. It is possible to export from the website, which will not include mbed CLI in the resulting zip.
+
+
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please delete the extra line.

File: docs/cont/adding_exporters.md [hunk 6]
```diff
@@ -1,18 +1,26 @@
...
 ## Export subsystem structure
 
 The export subsystem is organized as a group of common code and a group of IDE or toolchain specific plugins.
 
 The **common code** is contained in four files:
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please change "four" to "three".

File: docs/cont/adding_exporters.md [hunk 7]
```diff
@@ -230,3 +238,89 @@ $(PROJECT).elf: $(OBJECTS) $(SYS_OBJECTS) $(LINKER_SCRIPT)
...
+inherit from the appropriate Makefile class, and call its generate method.
+Taking Eclipse + Make using GCC_ARM as an example, your exporter will look like:
+
+```python
+class EclipseGcc(GccArm):
+    NAME = "Eclipse-GCC-ARM"
+```
+
+Your generate method will look similar to:
+
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please delete extra line.

File: docs/cont/adding_exporters.md [hunk 8]
```diff
@@ -230,3 +238,89 @@ $(PROJECT).elf: $(OBJECTS) $(SYS_OBJECTS) $(LINKER_SCRIPT)
...
+There are several paths forward that can lead to an easily maintained exporter:
+ - Specialize or alias the GNU ARM Eclipse exporter.
+ - Specialize or alias the Eclipse + Make exporter.
+ - Specialize the Make exporter.
+ 
+### GNU ARM Eclipse
+
+If your IDE uses Eclipse and uses the GNU ARM Eclipse plugin, then you should 
+specialize or alias your exporter with the generic GNU ARM Eclipse.
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Note: Throughout the document, a lot of the text gets split onto separate lines. [Example: line break between "should" and "specialize"] Is this preventable?

File: docs/cont/adding_exporters.md [hunk 9]
```diff
@@ -230,3 +238,89 @@ $(PROJECT).elf: $(OBJECTS) $(SYS_OBJECTS) $(LINKER_SCRIPT)
...
+
+If your IDE uses Eclipse and does not use the GNU ARM Eclipse plugin, you
+can use the "Unmanaged makefile" Eclipse exporter classes, `EclipseGcc`, 
+`EclipseArmc5` and `EclipseIar`. Much like the GNU ARM Eclipse section, you may 
+decide to alias or specialize.
+
+### Make
+
+If your IDE is not Eclipse based but can still use a Makefile, then you can 
+specialize the Makefile exporter. Specializing the Makefile is actually how we
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Query: Who is "we" in this sentence? ARM mbed?

File: docs/cont/adding_exporters.md (hunk 4 above)
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Query: When is "KDS" supposed to be capitalized, and when is it supposed to be lower case?

File: docs/cont/adding_exporters.md [hunk 10]
```diff
@@ -230,3 +238,89 @@ $(PROJECT).elf: $(OBJECTS) $(SYS_OBJECTS) $(LINKER_SCRIPT)
...
+## Suggested implementation
+
+There are several paths forward that can lead to an easily maintained exporter:
+ - Specialize or alias the GNU ARM Eclipse exporter.
+ - Specialize or alias the Eclipse + Make exporter.
+ - Specialize the Make exporter.
+ 
+### GNU ARM Eclipse
+
+If your IDE uses Eclipse and uses the GNU ARM Eclipse plugin, then you should
```
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please delete "you should" to match the format of the paragraphs after this one.

**Comment by screamerbg** [comment]
This is great! Thanks, Jimmy

File: docs/cont/adding_exporters.md (hunk 3 above)
Comment (AnotherButler) [inline docs/cont/adding_exporters.md]: Please delete "should", and change the mood of the bullets to reflect this. [Take > Takes, Use > Uses, Have > Has, Not call > Does not call]

File: docs/cont/adding_exporters.md (hunk 4 above)
Comment (theotherjimmy) [inline docs/cont/adding_exporters.md]: KDS, should be capitalized when referring to the IDE, and lower case when referring to the exporter.

**Comment by toyowata** [comment]
Hi @theotherjimmy 
This is nice.

File: docs/cont/adding_exporters.md [hunk 11]
```diff
@@ -1,18 +1,25 @@
 # Adding exporters 
 
-This is a guide for adding exporters to the mbed-os tools. It covers the structure of the export subsystem and the individual exporter.
+This is a guide for adding exporters to the mbed-os tools. First, this document describes what an exporter is and what rules it follows. Then, it covers the structure of the export subsystem and the individual exporter. Finally, this document gives some implementation suggestions.
```
Comment (iriark01) [inline docs/cont/adding_exporters.md]: mbed OS. That's one pound to the swear jar.

File: docs/cont/adding_exporters.md [hunk 12]
```diff
@@ -1,18 +1,25 @@
...
 
 <span class="notes">**Note:** All paths are relative to [https://github.com/ARMmbed/mbed-os/](https://github.com/ARMmbed/mbed-os/).</span>
 
+## What an exporter is
+
+An exporter is a Python plugin to the mbed OS tools that convert a project using mbed CLI into one specialized for a particular IDE. For the best user experience, an exporter:
```
Comment (iriark01) [inline docs/cont/adding_exporters.md]: On docs.mbed, I don't think the list will be displayed unless there's a blank line between it and the paragraph above it (after "an exporter:")

Comment (iriark01) [inline docs/cont/adding_exporters.md]: converts

Comment (theotherjimmy) [inline docs/cont/adding_exporters.md]: Alright, I'll add that line.

File: docs/cont/adding_exporters.md (hunk 11 above)
Comment (theotherjimmy) [inline docs/cont/adding_exporters.md]: Ah, old text. I'll update it.

**Pull request merged**